tests/baselines/reference/classStaticBlock24(module=amd).js

```diff
@@ -0,0 +1,24 @@
+//// [classStaticBlock24.ts]
+export class C {
+  static x: number;
+  static {
+    C.x = 1;
+  }
+}
+
+
+//// [classStaticBlock24.js]
+define(["require", "exports"], function (require, exports) {
+    "use strict";
+    exports.__esModule = true;
+    exports.C = void 0;
+    var C = /** @class */ (function () {
+        function C() {
+        }
+        return C;
+    }());
+    exports.C = C;
+    (function () {

```

Comment:
This is what I was concerned about and why I indicated we should add the test. One of my arguments for `static {}` was that it ensured the class declaration was fully initialized before its export binding was instantiated, so that you don't see a partially initialized class in a circular import scenario. Its fine for now, since we also incorrectly emit static field initializers after the binding is instantiated. The export binding issue can be investigated in a separate PR.